File: xiaodudou-server/src/main/java/ai/xiaodudou/module/feedback/entity/Feedback.java
package ai.xiaodudou.module.feedback.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Map;

@Data
@TableName(value = "t_feedback", autoResultMap = true)
public class Feedback implements Serializable {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private Long userId;
    private String content;
    private String contact;

    /** general / bug / suggestion / business */
    private String category;

    /** PENDING / PROCESSING / RESOLVED / IGNORED */
    private String status;

    private String reply;
    private LocalDateTime repliedAt;

    @TableField(typeHandler = JacksonTypeHandler.class)
    private Map<String, Object> clientInfo;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;

    public static final String STATUS_PENDING = "PENDING";
}

File: xiaodudou-miniapp/src/api/feedback.ts
import { http } from '../utils/request'

export interface FeedbackItem {
  id: number | string
  content: string
  contact?: string
  category: string
  status: 'PENDING' | 'PROCESSING' | 'RESOLVED' | 'IGNORED'
  reply?: string
  repliedAt?: string
  createdAt: string
}

export interface FeedbackSubmitReq {
  content: string
  contact?: string
  category?: 'general' | 'bug' | 'suggestion' | 'business'
  clientInfo?: Record<string, unknown>
}

export const feedbackApi = {
  submit(data: FeedbackSubmitReq) {
    return http.post<{ feedbackId: number; status: string; message: string }>('/api/v1/feedback', data as Record<string, unknown>)
  },
  myList() {
    return http.get<FeedbackItem[]>('/api/v1/feedback/my')
  }
}
